runtime/src/bpftime_shm_internal.cpp

```diff
@@ -484,8 +485,27 @@ int bpftime_shm::add_bpf_link(int fd, struct bpf_link_create_args *args)
 		errno = EBADF;
 		return -1;
 	}
-	return manager->set_handler(fd, bpftime::bpf_link_handler(*args),
-				    segment);
+	int result = manager->set_handler(
```

Comment:
Is it better to place it somewhere else for abstraction? Shall we put the attachment related code in the runtime here?

Reply:
uprobe_multi was attached in a very special way, which is different to all attaches using perf event. It fills attach targets in the bpf link, and does not use perf event or other targets(i.e target_fd would be 0 when attaching a uprobe multi). So now we don't have a suitable way to handle such abstractions. Maybe in the future when we rewritting shm client, we could split creating perf events for a uprobe multi to a individual function in the runtime